### ISSUE TITLE
defer file.close is a bad pattern

### ISSUE THREAD

**Issue by Macilias** [mmcgrana/gobyexample#246]
according to https://www.joeshaw.org/dont-defer-close-on-writable-files/ one should not use defer file.close(), its a bad pattern

I would suggest to change the last function of file defer.go to:

```
func closeFile(f *os.File) {
	fmt.Println("closing")
	err := f.Close()
	if err != nil {
		_, _ = fmt.Fprintf(os.Stderr, "error: %v\n", err)
		os.Exit(1)
	}
}
```

**Comment by eliben** [comment]
Sounds like a good idea to me. Would you like to send a PR?

**Comment by Macilias** [comment]
sure, here you are: https://github.com/mmcgrana/gobyexample/pull/248

**Comment by eliben** [comment]
Fixed by #248, thanks!
